feat!: switch to EKS managed node groups by default (#17)
camptocamp/devops-stack-module-cluster-eks: README.adoc

A https://devops-stack.io/[DevOps Stack] module to deploy a Kubernetes cluster on Amazon Web Services EKS.

This module creates https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html[EKS managed nodes] by default, bug also supports https://docs.aws.amazon.com/eks/latest/userguide/worker.html[self-managed node groups] by setting the variable `use_self_managed_node_groups` to `true`. It uses the Terraform https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest[module "eks" by AWS] to manage the cluster itself.

The `node_groups` variable is a map of objects. Each value of the map creates a node group referenced by its label. The value object can take any input accepted by the https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest/submodules/eks-managed-node-group[eks-managed-node-group] or https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest/submodules/self-managed-node-group[self-managed-node-group] submodule.

> Here is an example that creates one node group with 3 instances of type `r5a.large`, attaches them to the main Load Balancer, and customizes the base EBS volume size to 100 GB:

Reviewer: A quick question: does the use of the EKS managed node groups change the example on lines 11 to 30?

Author: No, the example remains valid as it is. The arguments accepted by the two types of node groups are largely the same.

This module needs a Route 53 DNS zone in the same AWS account to create a wildcard CNAME record that points to the Network LB. The DNS zone must be passed in the `base_domain` variable. This record is used by other DevOps Stack modules as default URLs for their applications.

> == Upgrading from versions earlier that 3.x

Reviewer: ```suggestion
== Upgrading from versions earlier than 3.x
```

> The versions 2.x and earlier of this module created https://docs.aws.amazon.com/eks/latest/userguide/worker.html[self-managed node groups] exclusively. Upgradeing to versions 3.x or later requires setting the variable `use_self_managed_node_groups` to `true` to avoid breaking existing clusters.

Reviewer: ```suggestion
The versions 2.x and earlier of this module created https://docs.aws.amazon.com/eks/latest/userguide/worker.html[self-managed node groups] exclusively. Upgrading to versions 3.x or later requires setting the variable `use_self_managed_node_groups` to `true` to avoid breaking existing clusters.
```

> Switching an existing cluster to EKS managed node groups is possible, but requires manually draining the existing nodes after the new node group is created and before the old one is destroyed (by apply terraform in multiple steps using `-target `) as well as manually adding the new node group to the Load Balancer.

Reviewer: ```suggestion
Switching an existing cluster to EKS managed node groups is possible, but requires manually draining the existing nodes after the new node group is created and before the old one is destroyed (by doing a `terraform apply -target <resource>` in multiple steps) as well as manually adding the new node group to the Load Balancer.
```
